app/controllers/iterations_controller.rb

```diff
@@ -79,118 +49,101 @@ def update_all
     # Iteration.task_graph_update response_hash
   end
 
-  def iteration_task
-    task = Task.find(params[:task_id])
-    @iteration = Iteration.find(params[:iteration_id])
-    if Task.no_update? task
-      redirect_to @iteration, notice: 'Uable to update as parent not fiinished.'
-    else
-      Task.update_status task
-      redirect_to @iteration
-    end
-  end
-
-  def iteration_task_reset
-    task = Task.find(params[:task_id])
-    @iteration = Iteration.find(params[:iteration_id])
-    task.reset_status
-    redirect_to @iteration
-  end
-
+  # delete task
   def delete_task
     task = Task.find(params[:task_id])
     @iteration = Iteration.find(params[:iteration_id])
     task.destroy
     redirect_to @iteration
   end
 
-  def create_task
-    @iteration = Iteration.find params[:iteration_id]
-    newtask = Task.new
-    newtask.updater_type = params[:updater_type]
-    newtask.title = params[:title]
-    newtask.task_status = "unstarted"
-    newtask.iteration_id = params[:iteration_id]
-    newtask.description = params[:description]
-    newtask.save
+  # Edit a single task
+  def edit_task
+    task = Task.find(params[:task_id])
+    @iteration = Iteration.find(params[:iteration_id])
+    task.description = params[:description]
+    task.title = params[:title]
+    task.save
     redirect_to @iteration
   end
 
-  def aggregate_tasks_graph
-    if current_user.is_student?
-      if current_user.project.nil?
-        redirect_to init_user_path current_user
-      else
-        redirect_to project_path current_user.project
+  # Finalize the assignment of template to specific project with
+  # starting and ending date
+  def apply_to
```

Comment:
This method has several cases, could you please specify what each case means?  It's probably better to refactor each if case into a private method, with informative names describing the case.